lib/connection.js

```diff
@@ -212,9 +212,13 @@ Connection.prototype._reconnect = function() {
     this.sasl_transport = undefined;
     this.transport = this.amqp_transport;
 
+    this.state.reconnect();
+    this._reset_remote_state();
+};
+
+Connection.prototype._reset_remote_state = function() {
```

Comment:
I reset the transport as well and then the manual connect worked out.

```
Connection.prototype._reset_remote_state = function() {
    //reset transport
    this.amqp_transport = new Transport(this.options.id, AMQP_PROTOCOL_ID, frames.TYPE_AMQP, this);
    this.sasl_transport = undefined;
    this.transport = this.amqp_transport;
    
    //reset remote endpoint state
    this.remote = {};
    //reset sessions:
    this.remote_channel_map = {};
    for (var k in this.local_channel_map) {
        this.local_channel_map[k]._reconnect();
    }
};
```

Reply:
Ok, I've pushed that change to the PR.